components-core/src/main/java/com/adyen/checkout/components/base/BaseConfigurationBuilder.kt

```diff
@@ -0,0 +1,57 @@
+package com.adyen.checkout.components.base
+
+import android.content.Context
+import com.adyen.checkout.components.util.ValidationUtils.isClientKeyValid
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.util.LocaleUtil
+import java.util.*
+
+/**
+ * Base constructor with the required fields.
+ *
+ * @param shopperLocale The Locale of the shopper.
+ * @param environment   The {@link Environment} to be used for network calls to Adyen.
+ * @param clientKey Your Client Key used for network calls from the SDK to Adyen.
+ */
+abstract class BaseConfigurationBuilder<ConfigurationT: Configuration>(
+        protected var mBuilderShopperLocale: Locale,
```

Comment:
These variables can be public. We only had setters in Java before, but allowing getters is not an issue.

This way you can use the direct koltin assignment in the other builders instead of the setters.

Reply:
Oh yeah, and rename to remove the `m` prefix.